src/GafferSceneUI/CropWindowTool.cpp

```diff
@@ -1018,3 +1020,20 @@ Box2f CropWindowTool::resolutionGate() const
 	}
 	return resolutionGate;
 }
+
+bool CropWindowTool::keyPress( const KeyEvent &event )
+{
+	if( const auto hotkey = Gaffer::Metadata::value<StringData>( this, "viewer:shortCut" ) )
+	{
+		if( event.key == hotkey->readable() && event.modifiers == KeyEvent::Modifiers::Alt )
+		{
+			const bool newState = !activePlug()->getValue();
+			activePlug()->setValue( newState );
+			if( m_cropWindowEnabledPlug )
+			{
+				m_cropWindowEnabledPlug->setValue( newState );
```

Comment:
I think this needs wrapping in an UndoScope.

Reply:
Good catch! That is added and squash into https://github.com/GafferHQ/gaffer/pull/5518/commits/92729599f03896bb38e0542a68396d737241ddce.

One small, I think, difference - I put the `UndoScope` just above the `m_cropWindowEnabledPlug->setValue()` line. I noticed undo didn't undo the `activePlug()->setValue()`, I think because it's not a descendant of `m_cropWindowEnabledPlug->ancestor<ScriptNode>()`?

I guess it doesn't matter for the functionality, but it seems better homed to put it above  `m_cropWindowEnabledPlug->setValue()` so as not to suggest it will undo `activePlug()->setValue()`.

Reply:
Yep, that placement for the UndoScope seems the most logical, and you're right as to why `activePlug()->setValue()` isn't undoable.